app/rtc_s1.py

```diff
@@ -147,6 +148,147 @@ def apply_slc_corrections(burst_in: Sentinel1BurstSlc,
     del band_out
 
 
+def calculate_layover_shadow_mask(burst_in: Sentinel1BurstSlc,
+                                 geogrid_in: isce3.product.GeoGridParameters,
+                                 path_dem: str,
+                                 path_scratch: str,
+                                 threshold_rdr2geo: float=1.0e-7,
+                                 numiter_rdr2geo: int=25,
+                                 extraiter_rdr2geo: int=10,
+                                 lines_per_block_rdr2geo: int=1000,
+                                 threshold_geo2rdr: float=1.0e-8,
+                                 numiter_geo2rdr: int=25,
+                                 dem_block_margin_geo2rdr: float=0.1):
+    '''
+    Generate the layover shadow mask and geodode the mask
+
+    Parameters:
+    -----------
+    burst_in: Sentinel1BurstSlc
+        Input burst
+    geogrid_in: isce3.product.GeoGridParameters
+        Geogrid to geocode the layover shawod mask in radar grid
+    path_dem: str
+        Path to the DEM
+    path_scratch: str
+        Path to the scratch directory
+    threshold_rdr2geo: float
+        Iteration threshold for rdr2geo
+    numiter_rdr2geo: int
+        Number of max. iteration for rdr2geo object
+    extraiter_rdr2geo: int
+        Extra number of iteration for rdr2geo object # TODO revise for better clarity
+    lines_per_block_rdr2geo: int
+        Lines per block for rdr2geo
+    threshold_geo2rdr: float
+        Iteration threshold for geo2rdr
+    numiter_geo2rdr: int
+        Number of max. iteration for geo2rdr object
+    dem_block_margin_geo2rdr: float
+        Block margin of DEM
+
+
+    Return:
+    -------
+    path_layover_shadow_mask_geogrid: str
+        Path to the geocoded layover shadow mask
+
+    '''
+
+    # TODO Consider using 'cfg' as argument to replace lots of arguments above
```

Comment:
Quite a lot of arguments (e.g. thresholds, # iterations, etc.) could be directly pulled from `cfg`.
I wonder what you think about just using `cfg` to replace them.

Reply:
Decided leave them as they are, so that we can easily reuse the function to somewhere else.